1.illumination_correction/qc_update/nbconverted/1.qc_report.py

```diff
@@ -312,3 +315,77 @@
 plt.savefig(output_directory / "qc_failure_rates_heatmap.png", dpi=500)
 plt.show()
 
+
+# ## Check only the images from healthy wells and see what is the biggest reason images are failing QC
+
+# In[10]:
+
+
+# Filter for healthy wells
+healthy_wells = ["B02", "B05", "B08", "B11"]
+healthy_df = combined_df[combined_df["Metadata_Well"].isin(healthy_wells)]
+
+# Get total FOVs per well
+total_per_well = (
+    healthy_df.groupby("Metadata_Well").size().reset_index(name="Total_FOVs")
+)
+
+# Prepare melted DataFrame
+melted = healthy_df.melt(
+    id_vars=["Metadata_Well"],
+    value_vars=qc_columns,
+    var_name="QC_Channel_Condition",
+    value_name="Failed",
+)
+
+# Only failed
+failed_melted = melted[melted["Failed"]].copy()
+
+# Split channel and condition
+failed_melted[["Channel", "Condition"]] = failed_melted[
+    "QC_Channel_Condition"
+].str.extract(r"(Orig\w+)_(Saturated|Blur)")
+
+# Count number failed per well/channel/condition
+failed_counts = (
+    failed_melted.groupby(["Metadata_Well", "Channel", "Condition"])
+    .size()
+    .reset_index(name="Failed_Count")
+)
+
+# Merge with total to get percentages
+percentages = failed_counts.merge(total_per_well, on="Metadata_Well")
+percentages["Percent_Failed"] = (
+    100 * percentages["Failed_Count"] / percentages["Total_FOVs"]
+)
+
+# Sort (optional)
+percentages = percentages.sort_values(["Metadata_Well", "Condition", "Channel"])
+
+# Show some data
+percentages.head()
+
+
+# In[11]:
+
+
+import seaborn as sns
+import matplotlib.pyplot as plt
```

Comment:
consider moving this at the top of your notebook to avoid imports during runtime.